services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/IslCostSpec.groovy

```diff
@@ -131,4 +131,48 @@ class IslCostSpec extends BaseSpecification {
             assert islUtils.getIslInfo(links, reverseIsl).get().state == IslChangeType.DISCOVERED
         }
     }
+
+    def "ISL(not BFD) is NOT FAILED earlier than discoveryTimeout is exceeded when connection is lost(not port down)"() {
+        given: "ISL going through a-switch and state of ISL"
+        def isl = topology.islsForActiveSwitches.find {
+            it.aswitch?.inPort && it.aswitch?.outPort && !it.bfd
+        } ?: assumeTrue("Wasn't able to find suitable ISL", false)
+
+        assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+
+        double waitTime = discoveryTimeout - (discoveryTimeout * 0.2)
+        double interval = discoveryTimeout * 0.2
+        def ruleToRemove = [isl.aswitch]
+        def reverseIsl = islUtils.reverseIsl(isl)
+
+        when: "Remove a one-way flow on an a-switch for simulating lost connection(not port down)"
+        lockKeeper.removeFlows(ruleToRemove)
+
+        then: "Status of ISL is not changed to FAILED until discoveryTimeout is exceeded"
+        Wrappers.timedLoop(waitTime) {
+            def links = northbound.getAllLinks()
+            assert islUtils.getIslInfo(links, isl).get().state == IslChangeType.DISCOVERED
+            assert islUtils.getIslInfo(links, reverseIsl).get().state == IslChangeType.DISCOVERED
+            sleep((interval * 1000).toLong())
+        }
+
+        and: "Status of ISL is changed to FAILED when discoveryTimeout is exceeded"
+        Wrappers.wait(WAIT_OFFSET) {
+            def links = northbound.getAllLinks()
+            assert islUtils.getIslInfo(links, isl).get().state == IslChangeType.FAILED
+            assert islUtils.getIslInfo(links, isl).get().actualState == IslChangeType.FAILED
+            assert islUtils.getIslInfo(links, reverseIsl).get().state == IslChangeType.FAILED
+            assert islUtils.getIslInfo(links, reverseIsl).get().actualState == IslChangeType.DISCOVERED
```

Comment:
great. A minor comment of why it acts like this in the code would be a plus (since we don't describe this in the step description it is good to document any unusual things)